ui/src/plugins/dev.perfetto.WidgetsPage/demos/nodegraph_demo.ts

```diff
@@ -1235,6 +1269,60 @@ export function NodeGraphDemo(): m.Component<NodeGraphDemoAttrs> {
             }
           });
         },
+        labels: store.labels,
+        selectedLabelIds: selectedLabelIds,
+        onLabelMove: (labelId: string, x: number, y: number) => {
+          updateStore((draft) => {
+            const label = draft.labels.find((l) => l.id === labelId);
+            if (label) {
+              label.x = x;
+              label.y = y;
+              console.log(`onLabelMove: ${labelId} to (${x}, ${y})`);
+            }
+          });
+        },
+        onLabelResize: (labelId: string, width: number) => {
+          updateStore((draft) => {
+            const label = draft.labels.find((l) => l.id === labelId);
+            if (label) {
+              label.width = width;
+              console.log(`onLabelResize: ${labelId} to width ${width}`);
+            }
+          });
+        },
+        onLabelTextChange: (labelId: string, text: string) => {
+          updateStore((draft) => {
+            const label = draft.labels.find((l) => l.id === labelId);
+            if (label) {
+              label.text = text;
+              console.log(`onLabelTextChange: ${labelId} to "${text}"`);
+            }
+          });
+        },
+        onLabelSelect: (labelId: string) => {
+          selectedLabelIds.clear();
```

Comment:
Why is there a different and separate selection mechanism for labels compared to nodes? Feels weird that you can select a node and a label at the same time.